libtiny_client/src/state.rs

```diff
@@ -232,6 +271,31 @@ impl StateInner {
                 }
             }
 
+            // Reply 477 when user needs to be identified with NickServ to join a channel
+            Reply { num: 477, params } => {
+                // Only try to automatically rejoin if nickserv_ident is configured
+                if self.nickserv_ident.is_some() {
+                    // Get channel name
+                    let channel = params.get(1).unwrap().clone();
+                    // Find channel in self.chans
+                    let idx = utils::find_idx(&self.chans, |c| c.name == channel).unwrap();
+                    // Retry joining channel if retries are available
+                    if self.chans[idx].retry_join() {
+                        let mut snd_irc_msg = snd_irc_msg.clone();
+                        // Spawn task and delay rejoin to give NickServ time to identify nick
+                        tokio::task::spawn_local(async move {
+                            tokio::time::delay_for(tokio::time::Duration::from(
+                                tokio::time::Duration::from_secs(10),
+                            ))
+                            .await;
+                            debug!("Attempting to re-join channel {}", channel);
```

Comment:
I wonder if we should generate a client event here (see `snd_ev`) so that we can show the progress in the UI. I think currently the UX is not great: if I just connected to a server where I want to join x, y, and z, and I joined x and y but not z, it's hard to see what's going on with z.

I think 477s should be shown in the channel tabs, so for example currently I see this in server tab:

```
helix.oftc.net: osa1_ #openjdk Cannot join channel (Need to be identified and verified to join this channel, '/msg NickServ help' to learn how to
register and verify.)
```

Instead we should create the channel and show this message there.

Once that's done we can also show the progress in the channel tab, e.g. something like: "Will try to re-join the channel in 10 seconds (attempt 1/3).".

What do you think?

Reply:
I like this idea. Is this like what you have in mind?: 
![image](https://user-images.githubusercontent.com/5448324/90865440-06b23600-e39b-11ea-9bb3-3f2750f64151.png)

On the server tab it still shows the same messages.

```
liquid.oftc.net: trevarj_ #vcs-home Cannot join channel (Need to be identified
and verified to join this channel, '/msg NickServ help' to learn how to register
and verify.)
```

Reply:
When the retry attempt is successful, the user will just see MOTD or nothing...